common/tasks/task.go

```diff
@@ -53,6 +53,21 @@ type (
 	}
 )
 
+func (left Key) CompareTo(right Key) int {
+	if left.FireTime.Before(right.FireTime) {
+		return -1
+	} else if left.FireTime.After(right.FireTime) {
+		return 1
+	}
+
+	if left.TaskID < right.TaskID {
```

Comment:
obviously I'm missing context here but it seems weird to consider one timer to be "before" the other just because their TaskID differ.  At the least I think this should be mentioned in the documentation of the function ... I would find this surprising behavior.

Reply:
this probably does not matter too much, as this compare only happen when fire time is the same. This uses FIFO if fire time is same.

Reply:
>obviously I'm missing context here but it seems weird to consider one timer to be "before" the other just because their TaskID differ.

timer & transfer / visibility / replication tasks all share the same task key definition
plus timers are sorted in DB (as well as in mem, in the future)